hazelcast/src/main/resources/hazelcast-config-4.1.xsd

```diff
@@ -650,6 +650,7 @@
                 <xs:restriction base="xs:string"/>
             </xs:simpleType>
         </xs:attribute>
+        <xs:attribute name="comparator" type="xs:string" use="optional"/>
```

Comment:
Unfortunately there's a lot more that needs to be done when introducing a new configuration option - examples, support for all declarative, programmatic and dynamic configuration, tests... Here's a [checklist](https://github.com/hazelcast/hazelcast/wiki/Checklist-for-Hazelcast-config-changes), please take a look. I can help you out with this if you get stuck.

Reply:
Sure I am aware of that and will take on all steps of the checklist.